### ISSUE TITLE
Enable rewiring rewired modules

### ISSUE THREAD

**Issue by TheSavior** [jhnns/rewire#77]
babel-plugin-rewire, rewireify, and rewire-global all hook require, injecting the functions into every module. 

When we use [getDefinePropertySrc](https://github.com/jhnns/rewire/blob/master/lib/getDefinePropertySrc.js) we run into problems on modules that pass through their dependencies unchanged. See this failing test on rewire-global: https://github.com/TheSavior/rewire-global/pull/13/files

The error we get is:
```
TypeError: Cannot redefine property: __get__
      at Function.defineProperty (native)
      at test/fixtures/keys-pass-through.js:7:8
      at Object.<anonymous> (test/fixtures/keys-pass-through.js:101:9)
      at Context.<anonymous> (test/tests/tests.js:63:5)
```

Before starting to use the upstream modules from rewire in rewireify and rewire-global, we got around this by setting the properties to writable: https://github.com/i-like-robots/rewireify/blob/master/lib/index.js#L37

Rewire [does not set them to writable](https://github.com/jhnns/rewire/blob/master/lib/getDefinePropertySrc.js#L17). I was looking through how the tests in rewire worked to try to add a failing test here and wasn't sure how to repro this since it is a bit of a different use case, but one we want to support.

**Comment by TheSavior** [comment]
Closing in favor of the PR
